spec/vector_spec.rb

```diff
@@ -254,6 +254,33 @@
             expect { @vector[:x, :one] }.to raise_error(IndexError)
           end
         end
+
+        context Daru::CategoricalIndex do
+          let(:idx1) { Daru::CategoricalIndex.new [:a, :b, :a, :a, :c] }
+          let (:dv1) { Daru::Vector.new 'a'..'e', index: @idx1 }
+
+          it "returns vector of elements belonging to given category" do
+            expect(@dv1[:a]).to eq(Daru::Vector.new(
+              ['a', 'c', 'd'], index: Daru::CategoricalIndex(
+                [:a, :a, :a])))
+          end
+
+          it "returns vector of elements belonging to given categories" do
+            expect(@dv1[:a, :c]).to eq(Daru::Vector.new(
+              ['a', 'c', 'd', 'e'], index: Daru::CategoricalIndex.new(
+                [:a, :a, :a, :c])))
```

Comment:
BTW, this case is already pretty good example when subject/its may be cleaner:
```ruby
subject { dv[:a, :c] }
it { is_expected.to be_a Daru::Vector }
its(:size) { is_expected.to eq 4 }
its(:to_a) { is_expected.to eq  ['a', 'c', 'd', 'e'] }
its(:index) { is_expected.to eq Daru::CategoricalIndex.new([:a, :a, :a, :c]) }
# or even
its(:index) { is_expected.to be_a Daru::CategoricalIndex }
its(:'index.to_a') { is_expected.to eq [:a, :a, :a, :c] }
```

Why? Because this way you can see clearer what part of specification is failing (for example, first you implement data part of sliced vector, then implement its index, and then do refactoring -- and only index order fails, and you see something like `expected [:a, :a, :a, :c], got [:a, :c]` instead of two large unequal vectors you need to scan by eyes to find what went sideways).

Though, I should say that such things are mostly the matter of personal preferences, so it is just a suggestion to consider. I will not bother you with those things too frequent :)